addons/account/models/account_payment.py

```diff
@@ -1047,9 +1047,13 @@ def action_post(self):
                     method_name=self.payment_method_line_id.name,
                     partner=payment.partner_id.display_name,
                 ))
+        payments_with_bank_outstanding_account = self.filtered(lambda pay: pay.outstanding_account_id.account_type == 'asset_cash')
+        other_payments = (self - payments_with_bank_outstanding_account)
+
+        payments_with_bank_outstanding_account.state = 'paid'
         # Avoid going back one state when clicking on the confirm action in the payment list view and having paid expenses selected
         # We need to set values to each payment to avoid recomputation later
-        self.filtered(lambda pay: pay.state in {False, 'draft', 'in_process'}).state = 'in_process'
+        other_payments.filtered(lambda pay: pay.state in {False, 'draft', 'in_process'}).state = 'in_process'
```

Comment:
Note that this change is actually useless since we already don't change the state of payments that are paid